staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -342,7 +342,7 @@ func (c *Controller) enqueueFoo(obj interface{}) {
 		utilruntime.HandleError(err)
 		return
 	}
-	c.workqueue.AddRateLimited(key)
```

Comment:
moreover, should call `AddRateLimited` for requeueing  if the sync handler returns error

Reply:
yes, of course, and this is exactly what sample-controller is doing, you can see this in [https://github.com/kubernetes/kubernetes/blob/0335867b564f4791ccca56f20b8d06226f991d15/staging/src/k8s.io/sample-controller/controller.go#L222](https://github.com/kubernetes/kubernetes/blob/0335867b564f4791ccca56f20b8d06226f991d15/staging/src/k8s.io/sample-controller/controller.go#L222)